.github/workflows/quality.yml

```diff
@@ -0,0 +1,95 @@
+name: Quality
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  linting:
+    name: Linting
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup
+        uses: ./.github/actions/ci
+
+      - name: Lint
+        run: pnpm lint
+
+  type-checking:
+    name: Type checking
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup
+        uses: ./.github/actions/ci
+
+      - name: TypeScript
+        run: pnpm typecheck:ts
+
+      - name: Flow
+        run: pnpm typecheck:flow
+
+  testing:
+    name: Testing
+    needs: [linting, type-checking]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup
+        uses: ./.github/actions/ci
+
+      - name: Build
+        run: pnpm build
+
+      - uses: oNaiPs/secrets-to-env-action@v1
+        with:
+          secrets: ${{ toJSON(secrets) }}
+
+      - name: Test (with coverage)
+        run: pnpm test:coverage:ci
+
+      - name: Integration tests
+        run: pnpm test:integration
+
+  regression-testing:
+    name: Regression Testing
+    needs: [testing]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        version: [18, 20]
```

Comment:
I think it will be better if we set this matrix to [20, 22], since 18 is now in maintenance [mode](https://nodejs.org/en/about/previous-releases).

Reply:
We test 22 in the regular testing flow. That's defined in the `.nvmrc` and what we develop with. v18 is not EoL and we officially still support it. Hence I would argue to also regression test for it. If we drop it, we should officially do so using the `engines` field and a major release. Keep in mind that with the latest [major release](https://github.com/commercetools/nodejs/releases/tag/%40commercetools%2Fsync-actions%407.0.0) we require >= Node.js v18 - prior we said to support v12 even.

Hope this makes sense. In a nutshell: let's not release major and deprecate Node.js v18 support in a migration to GitHub Actions.

Reply:
Sure, this make sense.